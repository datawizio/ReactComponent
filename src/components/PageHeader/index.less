@import "~antd/es/style/themes/default.less";
@import "../../styles/variables";

.page-header {
  display: flex;
  align-items: center;
  margin-bottom: @margin-sm;
  & > * {
    margin-right: @margin-xs;
    &:last-of-type {
      margin-right: 0;
    }
  }

  .page-header-title {
    color: @page-title-color;
    font-size: 20px;
    text-transform: uppercase;
    flex: 1;
  }
}
