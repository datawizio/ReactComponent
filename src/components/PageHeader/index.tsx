import React from "react";
import { ArrowLeftOutlined } from "@ant-design/icons";

import "./index.less";

interface PageHeaderProps {
  /**
   * Page title
   */
  title: string;

  onBack?: () => void;
}

const PageHeader: React.FC<PageHeaderProps> = ({ title, onBack, children }) => {
  return (
    <div className="page-header">
      {onBack && (
        <span className="page-header-back">
          <ArrowLeftOutlined onClick={onBack} />
        </span>
      )}
      <span className="page-header-title">{title}</span>
      {children}
    </div>
  );
};

export default PageHeader;
