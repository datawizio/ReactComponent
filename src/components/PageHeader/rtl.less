@import "../../styles/variables";

.rtl {
  .page-header {
    & > * {
      margin-right: unset;
      margin-left: @margin-xs;

      &:last-of-type {
        margin-right: unset;
        margin-left: 0;
      }
    }
  }
}