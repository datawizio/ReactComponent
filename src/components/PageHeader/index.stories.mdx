import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import PageHeader from "./index";
import Button from "../Button";

<Meta title="Components/PageHeader" component={PageHeader} />

# PageHeader

Компонент предназначен для вывода хидера страницы.

## Basic

<Preview>
  <Story name="Basic">
    <PageHeader title="Title" />
  </Story>
</Preview>

## Header with controls

<Preview>
  <Story name="Header with controls">
    <PageHeader title="Title">
      <Button>Action 1</Button>
      <Button type="primary">Action 2</Button>
    </PageHeader>
  </Story>
</Preview>

## Props

<Props of={PageHeader} />
