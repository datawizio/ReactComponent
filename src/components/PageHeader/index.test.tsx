import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import PageHeader from "./index";

const mockProps = {
  title: "Header title"
};

const setUp = (props?) => mount(<PageHeader {...props} />);

describe("PageHeader component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("PageHeader rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("PageHeader with  correct className", () => {
    expect(component.find(".page-header").length).toBeTruthy();
    expect(component.find(".page-header-back").length).toBeFalsy();
  });

  it("PageHeader render back button", () => {
    const wrapper = setUp({ ...mockProps, onBack: jest.fn() });
    expect(wrapper.find(".page-header-back").length).toBeTruthy();
  });

  it("PageHeader render correct inner html", () => {
    expect(component.find(".page-header-title").html()).toContain(
      mockProps.title
    );
  });
});
