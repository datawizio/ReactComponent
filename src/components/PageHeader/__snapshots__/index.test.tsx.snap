// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component PageHeader rendered correctly 1`] = `
<PageHeader
  title="Header title"
>
  <div
    className="page-header undefined"
  >
    <span
      className="page-header-title"
    >
      Header title
    </span>
  </div>
</PageHeader>
`;
